Create 2024-08-05.md (#811)
solid/specification: meetings/2024-08-05.md

# W3C Solid Community Group: Weekly

* Date: 2026-08-05T14:00:00Z
* Call: https://meet.jit.si/solid-cg
* Repository: https://github.com/solid/specification

## Chair

* [elf Pavlik](https://elf-pavlik.hackers4peace.net)

## Present

* Alain Bourgeois
* [Erich Bremer](https://ebremer.com/id/erich)
* Roberto S.K. Breitman
* Rui Zhao
* [Jesse Wright](https://jeswr.org/)

### Regrets

* 

## Scribe

> * 

Reviewer: Who scribed?

Author: i scribed a little bit, others assisted as well, there's also a lot missing once discussions started.

> * 

Author: ```suggestion
* sporadic and random
```

> * sporadic and random

Reviewer: Accountability is a virtue.

Reviewer 2: Scribing is very near a thankless job, and comments like this do not make it feel any better for those who receive them.

I, and I am sure others, will thank you to volunteer your scribing talents in future.

After all, volunteerism is a virtue.

Reviewer: @TallTed, appreciate your comment. And indeed, scribing is usually taken for granted and volunteerism is a virtue. And, I want to apologize to @elf-pavlik for that rather passive-aggressive comment of mine. 

That said, stating who took the minutes is a formality with a purpose, namely to create a bit of accountability within the CG, which I see as part of the duty of a CG chair.

However on,
> I, and I am sure others, will thank you to volunteer your scribing talents in future.

I am genuinely unsure how to take this statement. I did not attend the meeting on 2026-08-05, so I could not scribe. I regularly offer my service as a scribe in CG and WG meetings I attend. When chairing CG meetings myself, I do scribe often myself. I plan to keep doing that in the future (as e.g. done in https://github.com/solid/specification/pull/812). That's why I am left confused here. Mind to clarify what you mean, @TallTed?

Author: Thank you both @TallTed and @uvdsl 
And yes @uvdsl is most likely the most regular scribe, including more often than not scribing and chairing the same meeting.
To clarify this specific meeting, I was intending it as less formal since many people are taking vacation. Later in the meeting Rui brought inteersting topics so we scrambled some notes together. 
I will give more attention to 
* https://github.com/w3c-cg/solid/issues/77

so that we always have that option on the table

Reviewer 2: I was more snarky than was called for, for which I apologize. I was offended by @uvdsl's tone and I didn't look at scribing statistics (which I don't think are tracked anywhere that is easily viewable), and I mistakenly thought that @uvdsl was speaking in a "do as I say, not as I do" fashion. My bad. Speaking for myself, I am a terrible primary scribe, so I very rarely volunteer, and I try to make up for that by patching the notes taken by other scribes during meetings and by submitting PRs and other corrections after the fact.

Perhaps in future, instead of either leaving the scribe identifier blank or making it something like "sporadic and random", we should be more clear about saying something like "due to low attendance and minimal focused agenda, this meeting turned into an ad hoc conversation, and we failed to take many notes". 

Making one or more AI scribe tools available will also help, but I must caution against treating these as complete _solutions;_ they are steadily improving, but the number of errors made by even the best AI scribe of which I am currently aware demand significant after-scribing corrective action by human overseers.

If we consistently use the same LLM, it may be helpful to continue their training by making them ingest both their initial output and the final rendition after we fully correct the documents ... but even then, there's no guarantee that the human output will be perfect, so we may wind up reinforcing the LLM's bad guesses. It's a significant challenge!

* [Join the W3C Solid Community Group](https://www.w3.org/community/solid/join), [W3C Account Request](http://www.w3.org/accounts/request), [W3C Community Contributor License Agreement](https://www.w3.org/community/about/agreements/cla/)
* [Solid Code of Conduct](https://github.com/solid/process/blob/main/code-of-conduct.md), [Positive Work Environment at W3C: Code of Conduct](https://www.w3.org/policies/code-of-conduct/)
* Operating principle for effective participation is to allow access across disabilities, across country borders, and across time. Feedback on tooling and meeting timing is welcome.
* If this is your first time, welcome! Please introduce yourself.

---

## Introductions

## Announcements

### OS Tools meeting - August 05th TODAY!
- https://www.w3.org/events/meetings/a6b0a3dc-30fe-4f08-ac74-f51ba5418d07
- https://lists.w3.org/Archives/Public/public-solid/2026Jul/0016.html

* eP: WASM, WASI, WIT, WAC, jco, Component Model

## Actions Review

* Michal to offer use case for application needs in LDO, regarding app capabilities
* Erich to check SPARQL 1.2 and HTTP QUERY status
    https://w3c.github.io/sparql-protocol/spec/#query-operation

>     > The query operation must be invoked with either the HTTP GET or HTTP POST method (or, if the SPARQL Protocol service supports such requests, the HTTP QUERY method [RFC10008]).
>     > The HTTP QUERY method [RFC10008] can be used with SPARQL queries. Its use should be similar to the existing definitions and guidance for using HTTP POST with SPARQL queries with no other SPARQL-specific requirements. In the following sections, where SPARQL query requests using HTTP POST are described, HTTP QUERY may also be considered as an acceptable though not required feature of the SPARQL 1.2 Protocol.

Reviewer 2: This was clearly meant to be two paragraphs. But Markdown requires an empty line for that.
```suggestion
    > The query operation must be invoked with either the HTTP GET or HTTP POST method (or, if the SPARQL Protocol service supports such requests, the HTTP QUERY method [RFC10008]).
    >
    > The HTTP QUERY method [RFC10008] can be used with SPARQL queries. Its use should be similar to the existing definitions and guidance for using HTTP POST with SPARQL queries with no other SPARQL-specific requirements. In the following sections, where SPARQL query requests using HTTP POST are described, HTTP QUERY may also be considered as an acceptable though not required feature of the SPARQL 1.2 Protocol.
```

* Jesse to setup ai diffing of the specs (Solid & LWS)
  https://github.com/jeswr/lws-solid-diff
  

## Topics

### Demos

### TPAC Session for Solid CG

https://github.com/w3c-cg/solid/issues/84

* eP: Please keep in mind registration fee, including on-line participations. [Fee waivers available](https://www.w3.org/news-events/tpac/2026/registration/#registration-fee-waiver)

### Test suite for Solid Apps?

* RZ: There is a test suite for solid servers, I'm talking about apps. This was bourght up as well few months ago. I found there is https://www.npmjs.com/package/molid package. It is a JS package implemented by Angelo. Have anyone used it before? I'm trying to test solid apps agains something.
* AB: What have been tested in SolidOS - ActivityStreams, the only area I've seen it.

> * ...: You can check github solidos activitystreams pane uses that test.

Reviewer 2: ```suggestion
* RZ: There is a test suite for Solid servers, I'm talking about apps. This was brought up as well few months ago. I found there is https://www.npmjs.com/package/molid package. It is a JS package implemented by Angelo. Has anyone used it before? I'm trying to test Solid apps against something.
* AB: What has been tested in SolidOS — ActivityStreams, the only area I've seen it.
* ...: You can check GitHub SolidOS — ActivityStreams pane uses that test.

> * eP: is it UI or API
> * ...: how lib developer uses that with their client?
> * RZ: For me the problem is that there are several libraries, they work but may have hidden bugs
> * ...: I've been using Soukai, I found a bug that does double patch. When I bring it up it was a problem with CSS or the interaction between them. I can only add timeouts or delays. Only because this was known, but there may be unknown behavior somewhere.
> * ...: Test is pretty much a mock solid server, it may make it easier to define where the issue is.

Reviewer 2: ```suggestion
* eP: Is it UI or API
* ...: How lib developer uses that with their client?
* RZ: For me the problem is that there are several libraries. They work but may have hidden bugs.
* ...: I've been using Soukai; I found a bug that does double patch. When I brought it up, it was a problem with CSS or the interaction between them. I can only add timeouts or delays. Only because this was known, but there may be unknown behavior somewhere.
* ...: Test is pretty much a mock Solid server. That may make it easier to discover where the issue is.
```

> * eP: https://github.com/hackers4peace/interop.space/issues/13
> * ...: We can also ask devs, like Jackson and Michal for their experience with LDO
> * JW: I have codebase called Solid SDK, I have two useful things there. create-solid-app which bootstraps dev env including configured CSS instance for local testing, also agent skills to do testing against the local server. I'm moving from using CSS to use my own server implementation Sparq. It's rust which you can port to wasm and has much better startup.
> * EB: I had student creating FUSE adater for Solid, I ported it to LWS
> * RZ: https://github.com/renyuneyun/solid-fuse

Reviewer 2: ```suggestion
* eP: https://github.com/hackers4peace/interop.space/issues/13
* ...: We can also ask devs, like Jackson and Michal, for their experience with LDO.
* JW: I have a codebase called Solid SDK; I have two useful things there. `create-solid-app` which bootstraps dev env including a configured CSS instance for local testing; also agent skills to do testing against the local server. I'm moving from using CSS to use my own server implementation, **Sparq**. It's rust which you can port to wasm, and has much better startup.
* EB: I had a student creating a FUSE adapter for Solid. I ported it to LWS.
* RZ: https://github.com/renyuneyun/solid-fuse
```

* 

### AI / Coding Agent resources for Solid, and alternative preferences

> * RZ: I recall Jesse shared some claude skills. I find that there is a link to Solid SDK https://github.com/jeswr/solid-sdk
> * ...: how is it to be used, how do i customise it?
> * JW: my motivation was - some of the skills were referencing custom packages, i moved all those packages and skills referencing them to Solid SDK. There is still AGENTS.md in the root but you should be able point your agent to it and it will discover all the skills
> * ...: If you would like to make changes just create PRs and Issues
> * RZ: It makes sense. If i want to use it I can add the skills / subfolders

Reviewer 2: ```suggestion
* RZ: I recall Jesse shared some claude skills. I find that there is a link to the Solid SDK — https://github.com/jeswr/solid-sdk
* ...: How is it to be used? How do I customise it?
* JW: My motivation was — some of the skills were referencing custom packages. I moved all those packages and skills referencing them to the Solid SDK. There is still `AGENTS.md` in the root, but you should be able point your agent to it, and it will discover all the skills.
* ...: If you would like to make changes, just create PRs and Issues.
* RZ: It makes sense. If I want to use it, I can add the skills/subfolders.
```

> * JW: https://github.com/jeswr/solid-sdk/blob/main/AGENTS.md
> * ...: Insturction for Local agent skills. It points to offline-first package. There is another sills folder solid-sd/skills There are skills for reactive-authentication and other external libs. Also type indexes etc.
> * RZ: Does it mean that I need to add solid-sdk repo as a skill?
> * JW: What I would do, I don't know the details how agents refer to skills. In my conversation I link to agents.md as url and tell agent to set iself up with those skills.

Reviewer 2: ```suggestion
* JW: https://github.com/jeswr/solid-sdk/blob/main/AGENTS.md
* ...: Instruction for Local agent skills. It points to `offline-first` package. There is another skills folder, `solid-sd/skills`. There are skills for `reactive-authentication` and other external libs. Also type indexes, etc.
* RZ: Does that mean that I need to add `solid-sdk` repo as a skill?
* JW: What I would do, I don't know the details how agents refer to skills. In my conversation, I link to `agents.md` as url, and tell agent to set itself up with those skills.
```

> * RZ: If I need to personalize some of the opinionated skills, how do i do it?
> * ...: I guess I just copy content and make my modifications and use that version.
> * JW: I rarely touch skill files myself, I will in a conversation when pattern appears, I would tell LLM to update related skills to adjust it to what I expect.

Reviewer 2: ```suggestion
* RZ: If I need to personalize some of the opinionated skills, how do i do it?
* ...: I guess I just copy content and make my modifications and use that version?
* JW: I rarely touch skill files myself. In a conversation when a pattern appears, I would tell LLM to update related skills to adjust it to what I expect.
```
